recipe2023/index.js

```diff
@@ -0,0 +1,53 @@
+let button = document.querySelector("button");
+
+let color = "pink";
+button.addEventListener("click", () => {
+  color.style.backgroundColor = "pink";
+});
+
+//DOM ON CHECKLIST
+
+// let fetch = document.getElementById(checklist);
+// //let's create a list item and store in a variable
+// let newList = document.createElement("li");
+
+// //lets update the text content of that list item
+// newListItem.textContent = "Sirracha";
+
+// //Add that ul as list child
+// document.querySelector("ul").appendChild(checklist);
+const foodList = [
+  "momos",
+  "garlic",
+  "green chillies",
+  "spring onion",
+  "light soy sauce",
+  "sesame",
+  "rice vinegar",
+];
+
+const myDiv = document.getElementById("checklist");
+
+let checkbox = document.createElement("input");
+
+function check(array) {
+  for (let i = 0; i < array.length; i++) {
+    let checkbox = document.createElement("input");
+
+    checkbox.type = "checkbox";
+    checkbox.name = "name";
+    checkbox.value = "value";
+    checkbox.id = "id";
```

Comment:
You want to be careful with setting `id` to just `"id"` here because you have 7 iterations in this loop, that means it will generate 7 different checkboxes all with the same ID of `id`, which is not valid. An ID needs to be unique on a page. A common technique for addressing this type of problem is to use the index as part of the ID; e.g. 

```
checkbox.id = `checkbox-${i}`
```

Additionally, you probably want the `value` here to equal `array[i]`. The value of a checkbox is what is sent when you submit a form. The label as you have below is correct and will correctly show the correct text but that information is not sent over via a form submission.